compiler/src/iree/compiler/Codegen/Common/TransformDialectStrategiesGPU.cpp

```diff
@@ -164,24 +170,19 @@ static void createReductionStrategyThreadDistribution(
 
   // Split the reduction into a parallel and combiner part, then tile the
   // parallel part and map it to a full warp so it works on vectors.
-  auto tileReduction = b.create<transform::TileReductionUsingScfOp>(
+  auto tileReduction = b.create<transform::TileReductionUsingForeachThreadOp>(
       pdlOperation, pdlOperation, pdlOperation, pdlOperation, gridReductionH,
-      b.getI64ArrayAttr({0, firstReductionSize}));
+      /*numThreads=*/
+      b.getI64ArrayAttr({0, firstReductionSize}),
+      /*tileSizes=*/b.getI64ArrayAttr({0, vectorTileSize}),
+      /*threadDimMapping=*/b.getArrayAttr(threadX));
+  Value blockParallelForeachThreadOp = tileReduction.getForeachThreadOp();
```

Comment:
We should do a round of adding more builders to upstream ops. I've left another TODO for something in the v2 commit.